docs/index.asciidoc

```diff
@@ -140,6 +140,35 @@ input {
 }
 ---------------------------------------------------------------------------------------------------
 
+==== Prepared Statements
+
+Using server side prepared statements can speed up execution times as the server optimises the query plan and execution.
+
+NOTE: Not all JDBC accessible technologies will support prepared statements.
+
+With the introduction of Prepared Statement support comes a different code execution path and some new settings. Most of the existing settings are still useful but there are several new settings for Prepared Statements to read up on.
+Use the boolean setting `use_prepared_statements` to enable this execution mode. Use the `prepared_statement_name` setting to specify a name for the Prepared Statement, this identifies the prepared statement locally and remotely and it should be unique in your config and on the database. Use the `prepared_statement_bind_values` array setting to specify the bind values, use the exact string `:sql_last_value` (multiple times if necessary) for the predefined parameter mentioned before. The `statement` (or `statement_path`) setting still holds the SQL statement but to use bind variables you must use the `?` character as a placeholder in the exact order found in the `prepared_statement_bind_values` array.
+
+NOTE: Building count queries around a prepared statement is not supported at this time and because jdbc paging uses count queries under the hood, jdbc paging is not supported with prepared statements at this time either. Therefore, `jdbc_paging_enabled`, `jdbc_page_size` settings are ignored when using prepared statements.
```

Comment:
Do you think it makes sense to actively warn users or even fail to start if that use these settings that they are ignored when using prepared statements? Or mention in the definitions of these values that they are ignored when `use_prepared_statement` is false

Reply:
> Or mention in the definitions of these values that they are ignored when use_prepared_statement is false

They are not ignored when use_prepared_statement is false


Perhaps in the near future one of you folks will add the COUNT enhancement. My gut feeling is that a separate count_statement setting will need to be added because we can't rely on Sequel generating the correct SQL to paginate prepared statements.

Given (IMO) that pagination can't work without adding the above setting and therefore can't be a transparent plugin upgrade for the user (i.e. pagination just starts working) then a hard validation fail is necessary.
/cc @colinsurprenant

Reply:
Yes, I meant `use_prepared_statement` is `true` 🤦‍♂